src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -6844,6 +6844,17 @@ final int peekTokenType() throws SQLServerException
 		return currentPacket.payload[payloadOffset] & 0xFF;
 	}
 
+
+	final short peekStatusFlag() throws SQLServerException {
+		// skip the current packet(i.e, TDS packet type) and peek into the status flag (USHORT)
+		if (payloadOffset + 3 <= currentPacket.payloadLength) {
+			short value = Util.readShort(currentPacket.payload, payloadOffset + 1);
+			return value;
+		}
+
+		return Util.readShort(readWrappedBytes(3), 1);
+	}
+	
```

Comment:
the method is called "peek", should we not use Util.readXXX() to get the payload in the case of "peek"? I guess with Util.readXXX() , it actually "pop" the data?